### PR TITLE
CLDR-15572 liberty: enable monitoring and metrics

### PR DESCRIPTION
**Pull request by srl295**
CLDR-15572

add the `/metrics` endpoint to the server for prometheus monitoring

### CODE REVIEW

File: tools/cldr-apps/src/main/liberty/config/server.xml [hunk 1]
```diff
@@ -1,15 +1,17 @@
 <server description="cldr-apps Liberty server">
 	<!-- tag::featureManager[] -->
 	<featureManager>
-		<feature>jsonb-1.0</feature>
+		<feature>jsonb-1.0</feature> <!-- 2.0 -->
```
Comment (btangmu) [inline tools/cldr-apps/src/main/liberty/config/server.xml:4]: what do the comments indicate?

**Review by btangmu — APPROVED** [review]

File: tools/cldr-apps/src/main/liberty/config/server.xml (hunk 1 above)
Comment (srl295) [reply tools/cldr-apps/src/main/liberty/config/server.xml]: Oops, typo.  Those are the 'latest i could find' trying to track down a version conflict.  (I wanted to make this unobtrusive)

Comment (srl295) [reply tools/cldr-apps/src/main/liberty/config/server.xml]: I've removed the comments.

**Comment by srl295** [comment]
@btangmu  something new, for you:

```
# TYPE application_getDashboardCount_total counter
# HELP application_getDashboardCount_total Number of dashboards computed
application_getDashboardCount_total 3
# TYPE application_getDashboard_rate_per_second gauge
application_getDashboard_rate_per_second 0.053733788492078015
# TYPE application_getDashboard_one_min_rate_per_second gauge
application_getDashboard_one_min_rate_per_second 0.10937943541274854
# TYPE application_getDashboard_five_min_rate_per_second gauge
application_getDashboard_five_min_rate_per_second 0.17543537056339645
# TYPE application_getDashboard_fifteen_min_rate_per_second gauge
application_getDashboard_fifteen_min_rate_per_second 0.1913534483098531
# TYPE application_getDashboard_mean_seconds gauge
application_getDashboard_mean_seconds 1.9039911088907466
# TYPE application_getDashboard_max_seconds gauge
application_getDashboard_max_seconds 2.640120203
# TYPE application_getDashboard_min_seconds gauge
application_getDashboard_min_seconds 1.629090847
# TYPE application_getDashboard_stddev_seconds gauge
application_getDashboard_stddev_seconds 0.41588297098128657
# TYPE application_getDashboard_seconds summary
# HELP application_getDashboard_seconds Time to fetch the Dashboard
application_getDashboard_seconds_count 3
application_getDashboard_seconds{quantile="0.5"} 1.698057901
application_getDashboard_seconds{quantile="0.75"} 1.698057901
application_getDashboard_seconds{quantile="0.95"} 2.640120203
application_getDashboard_seconds{quantile="0.98"} 2.640120203
application_getDashboard_seconds{quantile="0.99"} 2.640120203
application_getDashboard_seconds{quantile="0.999"} 2.640120203
```

count and timing metrics on getDashboard() … just by adding these two lines:

```java
    @Counted(name = "getDashboardCount", absolute = true, description = "Number of dashboards computed")
     @Timed(absolute = true, name = "getDashboard", description = "Time to fetch the Dashboard")
   getDashboard(…)
```

File: tools/cldr-apps/src/main/java/org/unicode/cldr/web/SurveyLog.java [hunk 2]
```diff
@@ -74,6 +74,16 @@ public static void setDir(File homeFile) {
...
+        countException(exception);
+    }
+
+    /**
+     * Count an exception in the SurveyMetrics
+     */
+    public static void countException(final Throwable exception) {
+        if (CookieSession.sm != null && CookieSession.sm.surveyMetrics != null) {
+            CookieSession.sm.surveyMetrics.countException(exception);
+        }
```
Comment (srl295) [inline tools/cldr-apps/src/main/java/org/unicode/cldr/web/SurveyLog.java:86]: jaxrs stuff ( i.e. the `.api` subpackage ) won't need this hackery, somethiing like:

```java
class whatever {
     @Inject
     SurveyMetrics surveyMetrics;
```

… will be sufficient.

Comment (srl295) [reply tools/cldr-apps/src/main/java/org/unicode/cldr/web/SurveyLog.java:86]: the above is only to allow calling from a static environment

**Comment by srl295** [comment]
also note with this change you can view the metrics yourself at http://127.0.0.1:9080/metrics/

**Comment by srl295** [comment]
this obviates a bunch of stuff including the separate Node.js prometheus scraper, and stuff in the `what=status` action that was recently re-added.  Will complete the rest of the work in CLDR-15040 which is actually for this purpose.  This PR is for the telemetry needed so we can start trapping these errs.

**Review by btangmu — APPROVED** [review]
this could provide a lot of insights!

**Pull request merged**